### PR TITLE
caching web fixes

### PR DESCRIPTION
**Pull request by idanr84**
- making sure we update the listener whenever we init a new container
- implementing webview sharing without persistent key

### CODE REVIEW

File: android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java [hunk 1]
```diff
@@ -20,46 +25,86 @@
...
-                eventDispatcher.dispatchEvent(new ShouldOverrideUrlLoadingEvent(getId(), SystemClock.nanoTime(), url, navigationType));
-            }
-        });
+    // This version of Init assumes there is PersistentKey implementation so most of the sharing instance logic is inside the setPersistentKey method
+    void secondVersioInit(final ThemedReactContext context, final RNWebViewManager viewManager){
+        rnWebView = new RNWebView(viewManager, context, this);
 
-        this.setLayoutParams(
+        //rnWebView.loadUrl("http://www.amazon.com");
+        setBackgroundColor(Color.GRAY);
```
Comment (noamco) [inline android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java]: please comment these lines

File: android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java [hunk 2]
```diff
@@ -20,46 +25,86 @@
...
         }
+    }
+
+    // This version of Init assumes there is no PersistentKey implementation and there's only one webView instance
+    void firstVersioInit(final ThemedReactContext context, final RNWebViewManager viewManager){
+        if (rnWebView == null){
+            rnWebView = new RNWebView(viewManager, context, this);
+
+            //rnWebView.loadUrl("http://www.amazon.com");
+            setBackgroundColor(Color.GRAY);
```
Comment (noamco) [inline android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java]: please comment these lines

File: android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 import android.os.Build;
+import android.util.Log;
+import android.view.View;
 import android.view.ViewGroup;
 import android.webkit.CookieManager;
 import android.webkit.ValueCallback;
 import android.webkit.WebSettings;
 import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.Button;
```
Comment (noamco) [inline android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java]: unused import

File: android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
+import android.graphics.Color;
 import android.os.Build;
+import android.util.Log;
+import android.view.View;
 import android.view.ViewGroup;
 import android.webkit.CookieManager;
 import android.webkit.ValueCallback;
 import android.webkit.WebSettings;
 import android.webkit.WebView;
+import android.webkit.WebViewClient;
```
Comment (noamco) [inline android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java]: unused import

File: android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java [hunk 5]
```diff
@@ -125,18 +172,24 @@ public void setInjectedJavaScript(@javax.annotation.Nullable String injectedJava
...
-                ((ViewGroup)webView.getParent()).removeView(webView);
-            }
-            addView(webView);
-        } else {
-            addView(rnWebView);
-            webViews.put(persistentKey,rnWebView);
-        }
-        invalidate();
-        requestLayout();
+//        if (webViews.containsKey(persistentKey)){
```
[the remark this answers is not in the document]
Comment (idanr84) [reply android/src/main/java/com/burnweb/rnwebview/ContainerWebView.java]: Yes. Like I said on top of the `firstVersioInit` - right now persistentKey is not used.

**Pull request merged**